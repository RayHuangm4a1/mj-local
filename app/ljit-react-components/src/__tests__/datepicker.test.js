import React from 'react';
import { shallow, mount, } from 'enzyme';
import DatePicker from '../components/datepicker';

jest.mock('antd/lib/date-picker');

describe('DatePicker', () => {
	it('handle default props', () => {
		const {
			className,
			isDisabled,
			format,
			style,
			onChange,
			onFocus,
			onBlur,
		} = DatePicker.defaultProps;

		expect(className).toBe('');
		expect(isDisabled).toBe(false);
		expect(format).toBe('YYYY/MM/DD');
		expect(style).toEqual({});
		expect(onChange).toBeDefined();
		expect(onChange).toBeInstanceOf(Function);
		expect(onFocus).toBeDefined();
		expect(onFocus).toBeInstanceOf(Function);
		expect(onBlur).toBeDefined();
		expect(onBlur).toBeInstanceOf(Function);
	});

	it('should renders correctly', () => {
		const value = new Date(2019, 1, 1);
		const wrapper = shallow(<DatePicker value={value} placeholder="mock-placeholder" />);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class ljit-date-picker', () => {
		const wrapper = shallow(<DatePicker />);

		expect(wrapper.hasClass('ljit-date-picker')).toEqual(true);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<DatePicker className={className} />);

		expect(wrapper.hasClass(className)).toEqual(true);
	});

	it('should handle change event', () => {
		const value = new Date(2019, 1, 1);
		const onChange = jest.fn();
		const event = {
			preventDefault: () => {},
			target: { value, },
		};
		const wrapper = shallow(<DatePicker value={value} onChange={onChange} />);

		wrapper.simulate('change', event);
		expect(onChange).toBeCalledWith(event);
	});

	it('should handle focus event', () => {
		const value = new Date(2019, 1, 1);
		const onFocus = jest.fn();
		const event = {
			preventDefault: () => {},
			target: { value, },
		};
		const wrapper = shallow(<DatePicker value={value} onFocus={onFocus} />);

		wrapper.simulate('focus', event);
		expect(onFocus).toHaveBeenCalledTimes(1);
		expect(onFocus).toBeCalledWith(event);
	});

	it('should handle blur event', () => {
		const value = new Date(2019, 1, 1);
		const onBlur = jest.fn();
		const event = {
			preventDefault: () => {},
			target: { value, },
		};
		const wrapper = shallow(<DatePicker value={value} onBlur={onBlur} />);

		wrapper.simulate('blur', event);
		expect(onBlur).toHaveBeenCalledTimes(1);
		expect(onBlur).toBeCalledWith(event);
	});

	it('should mount in a full DOM', () => {
		const className = 'mock-class';
		const isDisabled = false;
		const defaultValue = new Date(2019, 1, 1);
		const value = new Date(2019, 1, 1);
		const format = 'YYYY-MM-DD';
		const placeholder = 'mock-placeholder';
		const style = { color: '#F00', };
		const onChange = () => {};
		const onFocus = () => {};
		const onBlur = () => {};

		const wrapper = mount(
			<DatePicker
				className={className}
				isDisabled={isDisabled}
				defaultValue={defaultValue}
				value={value}
				format={format}
				placeholder={placeholder}
				style={style}
				onChange={onChange}
				onFocus={onFocus}
				onBlur={onBlur}
			/>
		);

		expect(wrapper.props().className).toBe(className);
		expect(wrapper.props().isDisabled).toEqual(isDisabled);
		expect(wrapper.props().defaultValue).toEqual(defaultValue);
		expect(wrapper.props().value).toEqual(value);
		expect(wrapper.props().format).toBe(format);
		expect(wrapper.props().placeholder).toBe(placeholder);
		expect(wrapper.props().style).toMatchObject(style);
		expect(wrapper.props().onChange).toEqual(onChange);
		expect(wrapper.props().onFocus).toEqual(onFocus);
		expect(wrapper.props().onBlur).toEqual(onBlur);
	});
});
