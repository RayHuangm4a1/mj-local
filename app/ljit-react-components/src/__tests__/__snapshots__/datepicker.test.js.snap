// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DatePicker
    className=""
    format="YYYY/MM/DD"
    isDisabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="mock-placeholder"
    style={Object {}}
    value={2019-01-31T16:00:00.000Z}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "ljit-date-picker",
      "defaultValue": undefined,
      "disabled": false,
      "dropdownClassName": "ljit-date-picker-popup",
      "format": "YYYY/MM/DD",
      "locale": Object {
        "lang": Object {
          "backToToday": "返回今天",
          "clear": "清除",
          "dateFormat": "YYYY年M月D日",
          "dateSelect": "选择日期",
          "dateTimeFormat": "YYYY年M月D日 HH时mm分ss秒",
          "dayFormat": "D日",
          "decadeSelect": "选择年代",
          "month": "月",
          "monthSelect": "选择月份",
          "nextCentury": "下一世纪",
          "nextDecade": "下一年代",
          "nextMonth": "下个月 (翻页下键)",
          "nextYear": "下一年 (Control键加右方向键)",
          "now": "此刻",
          "ok": "确 定",
          "placeholder": "请选择日期",
          "previousCentury": "上一世纪",
          "previousDecade": "上一年代",
          "previousMonth": "上个月 (翻页上键)",
          "previousYear": "上一年 (Control键加左方向键)",
          "rangePlaceholder": Array [
            "开始日期",
            "结束日期",
          ],
          "timeSelect": "选择时间",
          "today": "今天",
          "weekSelect": "选择周",
          "year": "年",
          "yearFormat": "YYYY年",
          "yearSelect": "选择年份",
        },
        "timePickerLocale": Object {
          "placeholder": "请选择时间",
        },
      },
      "onBlur": [Function],
      "onChange": [Function],
      "onFocus": [Function],
      "onOk": [MockFunction],
      "onOpenChange": [MockFunction],
      "placeholder": "mock-placeholder",
      "popupStyle": Object {},
      "showTime": undefined,
      "style": Object {},
      "transitionName": "slide-up",
      "value": 2019-01-31T16:00:00.000Z,
    },
    "ref": null,
    "rendered": null,
    "type": [MockFunction],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "ljit-date-picker",
        "defaultValue": undefined,
        "disabled": false,
        "dropdownClassName": "ljit-date-picker-popup",
        "format": "YYYY/MM/DD",
        "locale": Object {
          "lang": Object {
            "backToToday": "返回今天",
            "clear": "清除",
            "dateFormat": "YYYY年M月D日",
            "dateSelect": "选择日期",
            "dateTimeFormat": "YYYY年M月D日 HH时mm分ss秒",
            "dayFormat": "D日",
            "decadeSelect": "选择年代",
            "month": "月",
            "monthSelect": "选择月份",
            "nextCentury": "下一世纪",
            "nextDecade": "下一年代",
            "nextMonth": "下个月 (翻页下键)",
            "nextYear": "下一年 (Control键加右方向键)",
            "now": "此刻",
            "ok": "确 定",
            "placeholder": "请选择日期",
            "previousCentury": "上一世纪",
            "previousDecade": "上一年代",
            "previousMonth": "上个月 (翻页上键)",
            "previousYear": "上一年 (Control键加左方向键)",
            "rangePlaceholder": Array [
              "开始日期",
              "结束日期",
            ],
            "timeSelect": "选择时间",
            "today": "今天",
            "weekSelect": "选择周",
            "year": "年",
            "yearFormat": "YYYY年",
            "yearSelect": "选择年份",
          },
          "timePickerLocale": Object {
            "placeholder": "请选择时间",
          },
        },
        "onBlur": [Function],
        "onChange": [Function],
        "onFocus": [Function],
        "onOk": [MockFunction],
        "onOpenChange": [MockFunction],
        "placeholder": "mock-placeholder",
        "popupStyle": Object {},
        "showTime": undefined,
        "style": Object {},
        "transitionName": "slide-up",
        "value": 2019-01-31T16:00:00.000Z,
      },
      "ref": null,
      "rendered": null,
      "type": [MockFunction],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
